### PR TITLE
[ENH] Creation of `interfaces` module

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR is a first step at solving #3014 :

- It creates a new `interfaces` module with an `fmriprep` submodule.
- It moves all fmriprep confounds loading utilities to `nilearn.interfaces.fmriprep` (no deprecation cycle since we haven't released since their addition)
- It updates the API documentations
- It updates the relevant examples

Example code:

The previous

```python
from nilearn.input_data import fmriprep_confounds
fmriprep_confounds(...)
```

becomes:

```python
from nilearn.interfaces.fmriprep import load_confounds
load_confounds(....)
```

### CODE REVIEW

**Comment by NicolasGensollen** [comment]
[Link to the API](https://8718-1235740-gh.circle-artifacts.com/0/dev/modules/reference.html#module-nilearn.interfaces) in the artifacts.

Pinging @htwangtw since you know this part of the code better than anybody ;)

File: nilearn/interfaces/fmriprep/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""
+The :mod:`nilearn.interfaces.fmriprep` module includes tools to preprocess
+neuro-imaging data and access :term:`fMRIPrep` generated confounds.
```
Comment (htwangtw) [inline nilearn/interfaces/fmriprep/__init__.py:3]:
```suggestion
neuroimaging data and access :term:`fMRIPrep` generated confounds.
```

**Review by htwangtw — COMMENTED** [review]
Just one thing, otherwise LGTM!

**Review by bthirion — APPROVED** [review]
LGTM. Thx !

**Review by htwangtw — APPROVED** [review]
LGTM!

**Comment by NicolasGensollen** [comment]
Thanks for the reviews!
Merging...

**Pull request merged**